operator/pkg/controllers/etcd/etcd.go

```diff
@@ -0,0 +1,346 @@
+}
+
+func etcdPeerSecretNameFor(clusterName string) string {
+	return fmt.Sprintf("%s-etcd-peer", clusterName)
+}
+
+func etcdLabelFor(clusterName string) map[string]string {
+	return map[string]string{
+		"app": etcdServiceNameFor(clusterName),
+	}
+}
+
+func namespacedName(namespace, obj string) types.NamespacedName {
+	return types.NamespacedName{Namespace: namespace, Name: obj}
+}
+
+func certListFor(controlPlane *v1alpha1.ControlPlane, caCert *x509.Certificate, caKey crypto.Signer) []*CertConfig {
+	return []*CertConfig{
+		etcdServerCertConfig(controlPlane, caCert, caKey),
+		etcdPeerCertConfig(controlPlane, caCert, caKey),
+	}
+}
+
+func etcdRootCACertConfig(controlPlane *v1alpha1.ControlPlane) *CertConfig {
+	return &CertConfig{
+		isCA: true,
+		name: etcdCASecretNameFor(controlPlane.ClusterName()),
+		Config: &certutil.Config{
+			CommonName: etcdRootCACommonName,
+		},
+	}
+}
+
+/*
+DNSNames contains the following entries-
+"localhost",
+<svcname>.<namespace>.svc.cluster.local
+<podname>
+<podname>.<svcname>.<namespace>.svc.cluster.local
+The last two entries are added for every pod in the cluster
+*/
+func etcdServerCertConfig(controlPlane *v1alpha1.ControlPlane, caCert *x509.Certificate, caKey crypto.Signer) *CertConfig {
+	config := &CertConfig{
+		caCert: caCert,
+		caKey:  caKey,
+		name:   etcdServerSecretNameFor(controlPlane.ClusterName()),
+		Config: &certutil.Config{
+			Usages:       []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+			CommonName:   "etcd",
+			Organization: []string{"kubernetes"},
+			AltNames: certutil.AltNames{
+				DNSNames: append(etcdPodAndHostnames(controlPlane),
+					etcdSvcFQDN(controlPlane.ClusterName(), controlPlane.Namespace),
+					"localhost"),
+				IPs: []net.IP{net.IPv4(127, 0, 0, 1)},
+			},
+		},
+	}
+	return config
+}
+
+/*
+DNSNames contains the following entries-
+"localhost",
+<svcname>.<namespace>.svc.cluster.local
+<podname>
+<podname>.<svcname>.<namespace>.svc.cluster.local
+The last two entries are added for every pod in the cluster
+*/
+func etcdPeerCertConfig(controlPlane *v1alpha1.ControlPlane, caCert *x509.Certificate, caKey crypto.Signer) *CertConfig {
+	config := &CertConfig{
+		caCert: caCert,
+		caKey:  caKey,
+		name:   etcdPeerSecretNameFor(controlPlane.ClusterName()),
+		Config: &certutil.Config{
+			Usages:       []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+			CommonName:   "etcd",
+			Organization: []string{"kubernetes"},
+			AltNames: certutil.AltNames{
+				DNSNames: append(etcdPodAndHostnames(controlPlane),
+					etcdSvcFQDN(controlPlane.ClusterName(), controlPlane.Namespace),
+					"localhost"),
+				IPs: []net.IP{net.IPv4(127, 0, 0, 1)},
+			},
+		},
+	}
+	return config
+}
+
+// Service name if <clustername>-etcd.<namespace>.svc.cluster.local
+func etcdSvcFQDN(clusterName, namespace string) string {
+	return fmt.Sprintf("%s.%s.svc.cluster.local", etcdServiceNameFor(clusterName), namespace)
```

Comment:
likewise. I think this is just `etcdServiceNameFor().svc`. Given that we're already scoped to etcd, consider removing etcd from all of these helpers.

I might also move all of these helpers to another file in this folder called `constants` or something